### PR TITLE
Johto map functionality

### PR DESCRIPTION
**Pull request by Aegyo**
No in game way of changing to johto just yet, but `MapHelper.moveToRoute(29,1)` will get you there.

- [x] Routes
- [x] Towns
- [x] Dungeons

### CODE REVIEW

File: src/scripts/pokemons/PokemonFactory.ts [hunk 1]
```diff
@@ -37,6 +37,15 @@ class PokemonFactory {
         return new BattlePokemon(name, id, basePokemon.type1, basePokemon.type2, maxHealth, route * 2, catchRate, exp, money, shiny);
     }
 
+    public static validRoute(route: number, region: GameConstants.Region): boolean {
```
Comment (dennism1997) [inline src/scripts/pokemons/PokemonFactory.ts]: Really nice

Comment (Aegyo) [reply src/scripts/pokemons/PokemonFactory.ts]: I was thinking of moving this into MapHelper maybe, it feels out of place in PokemonFactory.

File: src/scripts/worldmap/MapHelper.ts [hunk 2]
```diff
@@ -18,10 +19,17 @@ class MapHelper {
     };
 
     public static accessToRoute = function (route: number, region: GameConstants.Region) {
-        if (!player.hasBadge(GameConstants.routeBadgeRequirements[region][route])) {
+        if (region == GameConstants.Region.johto) { // Unlock all of johto for testing
+            return true
+        }
```
Comment (Ishadijcks) [inline src/scripts/worldmap/MapHelper.ts]: Now that all the requirements are added this should no longer be necessary right?

Comment (Aegyo) [reply src/scripts/worldmap/MapHelper.ts]: I don't think routeRequirements has been filled out yet.

File: src/scripts/worldmap/MapHelper.ts [hunk 3]
```diff
@@ -60,6 +68,7 @@ class MapHelper {
 
     public static accessToTown(townName: string): boolean {
         let town = TownList[townName];
+        if (!town) { return false }
```
[the remark this answers is not in the document]
Comment (Aegyo) [reply src/scripts/worldmap/MapHelper.ts]: I don't think this is needed anymore, I was just using it to avoid a bunch of errors while building the map.

**Comment by Ishadijcks** [comment]
We need images for the new cities.

**Comment by Aegyo** [comment]
Where did you get the Kanto city and dungeon images from?

**Comment by Ishadijcks** [comment]
Spriters-resource

**Pull request merged**